src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] innerArray = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (size == innerArray.length) {
+            grow();
+        }
+        innerArray[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            T[] arr = (T[]) new Object[size - index];
+            if (size > index) {
+                System.arraycopy(innerArray, index, arr, 0, size - index);
+            }
+            innerArray[index] = value;
+            if (size + 1 > innerArray.length) {
+                grow();
+            }
+            System.arraycopy(arr, 0, innerArray, index + 1, size - index);
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
 
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= 0 && index < size) {
+            return innerArray[index];
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index >= 0 && index < size) {
+            innerArray[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= 0 && index < size) {
+            T item = innerArray[index];
+            System.arraycopy(innerArray, index + 1, innerArray, index, size - index - 1);
+            size--;
+            return item;
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = findIndex(element);
+        if (index >= 0) {
+            return remove(index);
+        } else {
+            throw new NoSuchElementException("Index: " + size + ", Size: " + size);
```

Comment:
The message in `NoSuchElementException` is misleading. It should indicate that the element was not found, rather than providing index and size information.